delivery/media/watermarking/lib/media-delivery-watermarking.js

```diff
@@ -0,0 +1,2134 @@
+    }
+    async validateShortToken(authToken, keys) {
+        if ("string" != typeof authToken) throw new Error("Invalid token type, expected string!");
+        if (isEmptyString(authToken)) throw new Error("Token cannot be empty!");
+        if (0 == keys.length || !keys.every((item => "string" == typeof item && item.trim().length > 0))) throw new Error("Invalid keys type, expected array of hex encoded string!");
+        const tokenParts = authToken.split(".");
+        if (4 !== tokenParts.length) throw new Error("Token malformed: invalid short token format, expected token with four sets of base64url encoded fields joined by a period ‘.’!");
+        const cacheKey = await buildKey([ authToken ].concat(keys)), v = Cache.getShortToken(cacheKey), clockTimestamp = Math.floor(Date.now() / 1e3);
+        if (v) {
+            if (clockTimestamp > v.exp + (this.wmOptions.clockTolerance || 0)) throw new Error("Token expired!");
+            return v;
+        }
+        const exp = parseInt(uint8ArrayToHex(base64url.decode(tokenParts[0], "Uint8Array")), 16);
+        if (clockTimestamp > exp + (this.wmOptions.clockTolerance || 0)) throw new Error("Token expired!");
+        const title = base64url.decode(tokenParts[1], "String"), wmid = parseInt(uint8ArrayToHex(base64url.decode(tokenParts[2], "Uint8Array")), 16), signHex = uint8ArrayToHex(base64url.decode(tokenParts[3], "Uint8Array"));
+        let isTokenVerified = !1;
+        for (const k of keys) {
+            const signature = sha256.exports.hmac(base16.decode(k, "Uint8Array"), (new TextEncoder$1).encode(`${tokenParts[0]}.${tokenParts[1]}.${tokenParts[2]}`));
+            if (uint8ArrayToHex(new Uint8Array(signature)).substring(0, 30) === signHex) {
+                isTokenVerified = !0;
+                break;
+            }
+        }
+        if (!isTokenVerified) throw new Error("Token signature verification failed!");
+        const shortToken = {
+            exp,
+            wmid,
+            title
+        };
+        return Cache.storeShortToken(cacheKey, {
+            exp,
+            wmid,
+            title
+        }), shortToken;
+    }
+    async validateToken(authToken, keys) {
+        if (!("string" == typeof authToken && !isEmptyString(authToken) || authToken instanceof Uint8Array && authToken.length > 0)) throw new Error("Invalid token type, expected non empty string or non empty Uint8Array!");
+        if (!Array.isArray(keys) || 0 == keys.length || !keys.every((item => "string" == typeof item && item.trim().length > 0))) throw new Error("Invalid keys type, expected array of hex or pem encoded strings!");
+        const wmJSON = {}, cacheKey = await buildKey([ authToken ].concat(keys)), v = Cache.getToken(cacheKey);
+        if (v) return wmJSON.header = v.header, wmJSON.payload = v.payload, this.validateClaims(wmJSON.payload), 
+        wmJSON;
+        if (this.wmOptions.tokenType === TokenType.CWT) {
```

Comment:
Do we expect customers to validate both JWTs and CWTs with the same EW? 

If not, we're wasting initialization time parsing the unused token type. I suggest splitting this into two separate libraries and telling the user to import the appropriate one.

Reply:
@evan-hughes : Customer can validate both token types if they want, as of now we do not have hard requirement as such though. 
The customer would configure the token type when creating the watermarking object. (i.e either tokenType: CWT | JWT). That instance will always validated the token based on the configured token type. 
e.g const watermarkingJWT = new Watermarking( { tokenType: JWT }).

Main reason i have exported JWT and CWT module from this library is because:
- Watermarking module depends on cbor-x library for side car processing and CWT module depends on cbor-x too. If i separate out CWT module from watermarking, we would end up having cbor-x code (via rollup ) in both Watermarking and CWT module which will increase the initialization time further. 
- Watermarking token validation also requires token payload  to be validated with watermarking rules along with general JWT/CWT token validation rules

Reply:
Fair enough. I doubt this would be sizeable enough to cause init timeouts on its own. 

> Watermarking module depends on cbor-x library for side car processing and CWT module depends on cbor-x too. If i separate out CWT module from watermarking, we would end up having cbor-x code (via rollup ) in both Watermarking and CWT module which will increase the initialization time further.

If the cbor-x module is kept in a separate file and imported from the Watermarking and CWT modules, Stormcloud should only parse it once.